services/user-service.js

```diff
@@ -119,139 +120,164 @@ class UserService {
     }
-    if (user.friends.includes(acceptFriend)) {
+    const userToAccept = await UserModel.findOne({ username : acceptFriend })
+    .populate("outcomingRequests", "username");
+
+    const isFriend = await FriendModel.findOne({$or: [
+      { requesterId: userId, receiverId : userToAccept.id },
+      { receiverId: userId, requesterId : userToAccept.id },
+    ]});
+    if (isFriend) {
       throw ApiError.BadRequest(`You are already friends with ${acceptFriend}`);
     }
-    const incomingRequests = user.incomingRequests;
-    const userIndex = incomingRequests.indexOf(acceptFriend);
-    if (userIndex < 0) {
-      throw ApiError.BadRequest(`User ${acceptFriend} did not send you request`);
-    }
 
-    user.incomingRequests.splice(userIndex, 1);
-    user.friends.push(acceptFriend);
+    await FriendModel.create({ requesterId : userToAccept.id, receiverId : userId });
+
+    user.incomingRequests.pull(request.id);
     await user.save();
 
-    const acceptFriendIndex = userToAccept.outcomingRequests.indexOf(user.username);
-    userToAccept.outcomingRequests.splice(acceptFriendIndex, 1);
-    userToAccept.friends.push(user.username);
+    const acceptFriendRequest = userToAccept.outcomingRequests.find(el => el.username === user.username);
+    userToAccept.outcomingRequests.pull(acceptFriendRequest.id);
     
     await userToAccept.save();
-    return user.friends;
+    return { message: `Request from ${acceptFriend} was accepted` }
   }
   /**
    * @param rejectFriend
    *   Username of a user you want to reject request from.
    */
   async rejectRequest(userId, rejectFriend){
-    const user = await UserModel.findById(userId);
-    const userToReject = await UserModel.findOne({ username : rejectFriend });
-    if (!user || !userToReject) {
-      throw ApiError.NotFound('User is not found');
+    const user = await UserModel.findById(userId)
+    .populate("incomingRequests friends.addresseeId", "username");
+
+    const request = user.incomingRequests.find(el => el.username === rejectFriend);
+    if (!request) {
+      throw ApiError.BadRequest(`User ${rejectFriend} did not send you request`);
     }
-    if (user.friends.includes(rejectFriend)) {
+    const userToReject = await UserModel.findOne({ username : rejectFriend })
+    .populate("outcomingRequests", "username");
+
+    const isFriend = await FriendModel.findOne({$or: [
+      { requesterId: userId, receiverId : userToReject.id },
+      { receiverId: userId, requesterId : userToReject.id },
+    ]});
+    if (isFriend) {
       throw ApiError.BadRequest(`You are already friends with ${rejectFriend}`);
     }
-    const incomingRequests = user.incomingRequests;
-    const userIndex = incomingRequests.indexOf(rejectFriend);
-    if (userIndex < 0) {
-      throw ApiError.BadRequest(`User ${rejectFriend} did not send you request`);
-    }
 
-    user.incomingRequests.splice(userIndex, 1);
+    user.incomingRequests.pull(request.id);
```

Comment:
if you have friend model that you don't need to have incomingRequests array in user schema, use friend model instead